### PR TITLE
DSET-927 Support argo rollouts

### PR DESCRIPTION
**Pull request by dlobue**
Customer  support issue - add support for the `Rollout` kind.

Also make this type of issue easier to debug in the future by adding code comments around the obscurely named `_k8s_d*` attributes, and change the log event from a debug message to a rate-limited warning.

DSET-927

### CODE REVIEW

File: scalyr_agent/builtin_monitors/kubernetes_monitor.py [hunk 1]
```diff
@@ -3226,8 +3226,20 @@ def __get_log_config_for_container(self, cid, info, k8s_cache, base_attributes):
                     rename_vars["deployment_name"] = controller.name
                     rename_vars["controller_name"] = controller.name
                     if self.__include_controller_info:
+                        # The following attributes are translated into server
+                        # fields, and the names are translated into the
+                        # following usable names once ingested.
```
Comment (jmakar-scalyr) [inline scalyr_agent/builtin_monitors/kubernetes_monitor.py:3231]: Very helpful, thanks for including this

File: scalyr_agent/monitor_utils/k8s.py [hunk 2]
```diff
@@ -892,9 +900,11 @@ def _get_controller_from_owners(self, k8s, owners, namespace, query_options=None
 
             if controller.parent_kind not in _OBJECT_ENDPOINTS:
                 global_log.log(
-                    scalyr_logging.DEBUG_LEVEL_1,
+                    WARNING,
```
Comment (jmakar-scalyr) [inline scalyr_agent/monitor_utils/k8s.py:903]: Generally should use logging.WARNING instead, up to you if you want to change it

File: scalyr_agent/monitor_utils/k8s.py [hunk 3]
```diff
@@ -103,6 +104,13 @@
         "list": Template("/apis/apps/v1/namespaces/${namespace}/statefulsets"),
         "list-all": "/apis/apps/v1/statefulsets",
     },
+    "Rollout": {
```
Comment (jmakar-scalyr) [inline scalyr_agent/monitor_utils/k8s.py:107]: Might be good to explicitly point out in a comment that this is for ArgoCD enabled clusters, I'll leave it up to you if you think it's worth adding

**Review by jmakar-scalyr — APPROVED** [review]

File: scalyr_agent/monitor_utils/k8s.py (hunk 2 above)
Comment (dlobue) [reply scalyr_agent/monitor_utils/k8s.py:903]: that is what I'm using, I'm importing WARNING from logging up at the top.

File: scalyr_agent/monitor_utils/k8s.py (hunk 3 above)
Comment (dlobue) [reply scalyr_agent/monitor_utils/k8s.py:110]: Added requested comments.

**Pull request merged**